spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaConsumerFactory.java

```diff
@@ -22,28 +23,53 @@
 import org.apache.kafka.clients.consumer.Consumer;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.Deserializer;
 
 /**
  * The {@link ConsumerFactory} implementation to produce a new {@link Consumer} instance
- * for provided {@link Map} {@code configs} on each {@link #createConsumer()}
+ * for provided {@link Map} {@code configs} and optional {@link Deserializer} {@code keyDeserializer},
+ * {@code valueDeserializer} implementations on each {@link #createConsumer()}
  * invocation.
  *
  * @param <K> the key type.
  * @param <V> the value type.
  *
  * @author Gary Russell
+ * @author Murali Reddy
  */
 public class DefaultKafkaConsumerFactory<K, V> implements ConsumerFactory<K, V> {
 
 	private final Map<String, Object> configs;
 
+	private Deserializer<K> keyDeserializer;
+
+	private Deserializer<V> valueDeserializer;
+
 	public DefaultKafkaConsumerFactory(Map<String, Object> configs) {
 		this.configs = new HashMap<>(configs);
 	}
 
+	public DefaultKafkaConsumerFactory(Map<String, Object> configs,
+			Deserializer<K> keyDeserializer,
+			Deserializer<V> valueDeserializer) {
+		this.configs = new HashMap<>(configs);
+		this.keyDeserializer = keyDeserializer;
+		this.valueDeserializer = valueDeserializer;
```

Comment:
Just because they aren't required options.
It is some style of our code.
If the option is required it goes to the ctor argument, otherwise it is setter do not confuse end user.
And again: with this new ctor I am forced to specify both `Deserializer`s , like `null` one of them, if I'm not interested in both. E.g. I would rely just on the default `Deserializer` for `key`, but would provide some specific for `value`.
So, I don't see reason to overhead it with optional ctor.